### PR TITLE
Add the ability to declare empty item operations

### PR DESCRIPTION
**Pull request by soyuka**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #3093 #3096 #3051 
| License       | MIT
| Doc PR        | https://github.com/api-platform/docs/pull/886

### CODE REVIEW

File: tests/Fixtures/TestBundle/Document/DisableItemOperation.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+namespace ApiPlatform\Core\Tests\Fixtures\TestBundle\Document;
+
+use ApiPlatform\Core\Action\NotFoundAction;
+use ApiPlatform\Core\Annotation\ApiResource;
+use Doctrine\ODM\MongoDB\Mapping\Annotations as ODM;
+
+/**
+ * DisableItemOperation.
+ *
+ * @author Antoine Bluchet <soyuka@gmail.com>
```
Comment (teohhanhui) [inline tests/Fixtures/TestBundle/Document/DisableItemOperation.php]: Unnecessary? 👻

File: tests/Fixtures/TestBundle/Document/DisableItemOperation.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ *
+ * @ApiResource(itemOperations={
```
Comment (teohhanhui) [inline tests/Fixtures/TestBundle/Document/DisableItemOperation.php]:
```php
/**
 * @ApiResource(
 *     collectionOperations={
 *         "get",
 *     },
 *     itemOperations={
 *         "get"={
 *             "controller"=NotFoundAction::class,
 *             "read"=false,
 *             "output"=false,
 *         },
 *     },
 * }
 */

File: src/Action/NotFoundAction.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Action;
+
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+/**
+ * Not found action.
+ *
+ * @author Antoine Bluchet <soyuka@gmail.com>
```
Comment (teohhanhui) [inline src/Action/NotFoundAction.php]: Unnecessary? 👻

File: tests/Fixtures/TestBundle/Document/DisableItemOperation.php (hunk 2 above)
Comment (soyuka) [reply tests/Fixtures/TestBundle/Document/DisableItemOperation.php]: I tried :laughing:

File: src/Action/NotFoundAction.php [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Action;
+
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+/**
+ * Not found action.
```
Comment (teohhanhui) [inline src/Action/NotFoundAction.php]: This summary adds nothing and should be removed, or reworded?

Comment (soyuka) [reply src/Action/NotFoundAction.php]: * Action that throws a NotFoundHttpException, useful to disable an operation.
 ?

File: src/Action/NotFoundAction.php (hunk 3 above)
Comment (soyuka) [reply src/Action/NotFoundAction.php]: it was to be consistent with the other actions

File: src/Action/NotFoundAction.php (hunk 4 above)
Comment (teohhanhui) [reply src/Action/NotFoundAction.php]: An action which always returns HTTP 404 Not Found. Useful for disabling an operation.

(Implementation details should be left out.)

File: tests/Fixtures/TestBundle/Document/DisableItemOperation.php [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Tests\Fixtures\TestBundle\Document;
+
+use ApiPlatform\Core\Action\NotFoundAction;
+use ApiPlatform\Core\Annotation\ApiResource;
+use Doctrine\ODM\MongoDB\Mapping\Annotations as ODM;
+
+/**
+ * DisableItemOperation.
```
Comment (teohhanhui) [inline tests/Fixtures/TestBundle/Document/DisableItemOperation.php]: Remove.

File: tests/Fixtures/TestBundle/Entity/DisableItemOperation.php [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Tests\Fixtures\TestBundle\Entity;
+
+use ApiPlatform\Core\Action\NotFoundAction;
+use ApiPlatform\Core\Annotation\ApiResource;
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * DisableItemOperation.
```
Comment (teohhanhui) [inline tests/Fixtures/TestBundle/Entity/DisableItemOperation.php]: Remove.

**Pull request merged**